HelpSource/Tutorials/QT/things_you_wanna_know.schelp

```diff
@@ -0,0 +1,61 @@
+title:: QT : Things you probably want to know
+summary:: Miscellaneous things you probably want to know about QT.
+categories:: Tutorials>QT
+
+section:: Rendering isn't always available
+
+SuperCollider provides some useful classes to automate processes. Some of them are associated to a link::Classes/Clock:: : mainly link::Classes/Routine:: and link::Classes/Task::.
+
+strong::QT rendering system is unavailable within the context of clock associated threads::. In other words, you cannot 'directly' modify a widget within a link::Classes/Routine:: or a link::Classes/Task::.
+
+Instead, you need to defer the modification by strong::encapsulating it in a deferred function::, that will be executed once SuperCollider resume its main context. Doing otherwise will crash the program :
+
+code::
+(
+var win = Window();
+var slider= Slider();
+var value = 0;
+
+var routine = Routine({
+	loop {
+		value = 101.rand / 100;
+		// Interacting with a widget from within a routine :
+		// instructions must be nested in a function, and deferred
+		{ slider.value_(value); }.defer;
+		0.5.wait;
+	};
+}).play;
+
+win.layout_(
+	VLayout()
+	.add(slider)
+);
+
+CmdPeriod.doOnce({
+	routine.stop;
+	win.close;
+});
+
+win.front;
+)
+::
+
+section:: QT limitations
+
+This remark mostly concerns debugging and issues understanding. Since strong::QT is a third-party program integrated within SuperCollider::, some of its limitations are unrelated to SuperCollider itself.
+
+As an example, QT5 link::Classes/QImage:: has a strict pixel limit and cannot exceed 32767x32767 pixels. This 'issue' being only related to QT, it is documented on QT sources/forums, but not on SuperCollider sources/forums. In other words, working with QT within the context of SuperCollider might also include using QTs documentation directly.
+
+section:: Pen usage and resource management
+
+The link::Classes/Pen:: class allows the user to define a custom code::drawFunc:: to a link::Classes/View::.
+
+link::Classes/Pen:: usage is a strong::two-steps process:: : the user strong::first 'describes' the shape:: he wants to draw (using methods such as code::fillRect:: or code::lineTo::), strong::then renders the shape::, using either code::fill::, code::stroke::, or code::fillStroke::.
```

Comment:
```suggestion
link::Classes/Pen:: usage is a strong::two-steps process:: : the user strong::first 'describes' the shape:: they want to draw (using methods such as code::fillRect:: or code::lineTo::), strong::then renders the shape::, using either code::fill::, code::stroke::, or code::fillStroke::.
```

We should not focus on a male-audience within the docs.

Reply:
Sorry about this. I understand and agree with this principle, but this preoccupation is mostly absent from the social context I'm leaving in, so thinking about it isn't an automatism yet.